### ISSUE TITLE
Can you explain how the bracket order works?

### ISSUE THREAD

**Issue by SethKitchen** [JoeBrewing/BracketGenerator#1]
I do

            OneVOne bracket = new OneVOne();
            bracket.OneVOneParticipants.Add(new OneVOneParticipant() { Name = "Joe", Seed = 1 });
            bracket.OneVOneParticipants.Add(new OneVOneParticipant() { Name = "Rob", Seed = 2 });
            bracket.OneVOneParticipants.Add(new OneVOneParticipant() { Name = "Nate", Seed = 3 });
            bracket.OneVOneParticipants.Add(new OneVOneParticipant() { Name = "Stephen", Seed = 4 });
            bracket.OneVOneParticipants.Add(new OneVOneParticipant() { Name = "Justin", Seed = 5 });
            bracket.OneVOneParticipants.Add(new OneVOneParticipant() { Name = "Blake", Seed = 6 });
            bracket.OneVOneParticipants.Add(new OneVOneParticipant() { Name = "Kamika", Seed = 7 });
            
            bracket.SetBracketOrder();

and I get

![image](https://user-images.githubusercontent.com/9014645/45276869-5cc98f80-b48a-11e8-96cd-bb2ea4d2dbe4.png)

Does this mean Rob should play Kamika, and Nate should play Blake? Why are they listed twice? Where are Stephen, Justin, and Joe?

**Comment by JoeBrewing** [comment]
Oh wow, I haven’t looked at that code in years!

I would have to refresh myself a little on the code to see what’s going on.
Have you looked at the unit tests in the test folder? There’s a lot of
bracket generation code in there.

I’d imagine that you probably want to be using GenerateBracket() instead of
SetBracketOrder().

Sorry the code is such a mess. I never really imagined anybody finding it.
I did it as a personal coding project to work with unit tests and generics.
How’d you find it? I’m curious.

**Comment by SethKitchen** [comment]
Yep the code I posted is straight from a unit test - I tried GenerateBracket(), but that doesn't produce anything as far as I can tell?

I don't think the code is a mess - it seems intuitive and that's why I pulled it just not sure where the actual bracket is generated. I just need a bracket generator for my C# application and didn't want to reinvent the wheel if it was already on github. You are one of 4 bracket generator programs in C# that I found -

**Comment by JoeBrewing** [comment]
The bracket is a bottom up generated tree. So the first round matches are at the bottom of the tree and the final is at the top of the tree. GenerateBracket populates the tree. If you look at the structure of FinalMatch after running GenerateBracket then you SHOULD have a fully populated bracket.

**Comment by SethKitchen** [comment]
ahh I see! Noice. Thanks

**Comment by SethKitchen** [comment]
ok now I pretty print the final match tree instead - same example - and it's still missing initial players on the list and repeating twice

![image](https://user-images.githubusercontent.com/9014645/45323625-5af6df00-b511-11e8-9488-86e691946b01.png)

![image](https://user-images.githubusercontent.com/9014645/45323544-1408e980-b511-11e8-8ef1-4d8feb6dc579.png)

**Comment by JoeBrewing** [comment]
Hmm...I'm honestly not sure what to tell you. Sorry.

It's been so long since I've worked on that code that I'd probably have to do a debug step through to see what's going on.
